version.sbt

```diff
@@ -1 +1 @@
-ThisBuild / version := "0.41-SNAPSHOT"
+ThisBuild / version := "1.0.0-SNAPSHOT"
```

Comment:
Also, i have [this](https://github.com/guardian/mobile-apps-api-models) project, should I be updating it to use this new plugin? (it's configured with my own sonatype username and password at the moment....)

Reply:
> we can run the release workflow and it will build/deploy version `1.0.0`? I suppose this means that, in the case where we're using this new plugin, the version.sbt file is the source of truth for the _next_ version?

That's right! Actually, I don't think that's a change from the existing behaviour for version numbers - the workflow hasn't changed things here. You can think of '-SNAPSHOT' as meaning '-SNAPSHOT-PREVIEW', so `1.0.0-SNAPSHOT` is actually a 'preview' release of `1.0.0` before it's actually finished and released. This is [standard Maven release versioning](https://maven.apache.org/guides/getting-started/index.html#what-is-a-snapshot-version) - we always work with the version number set to a snapshot of what the _next_ release will be.

> I created [this](https://github.com/guardian/mobile-apps-api-models) project, should I be updating it to use this new plugin? (it's configured with my own sonatype username and password at the moment....)

Well, at the moment, [gha-scala-library-release-workflow](https://github.com/guardian/gha-scala-library-release-workflow) is in an early-feedback phase - I'd like it tried out on just a small number of repositories, so people can give me feedback to incorporate before we update all the Scala library repos at the Guardian!

If you'd like to try out updating https://github.com/guardian/mobile-apps-api-models to work with the new flow, that would be absolutely wonderful, you'd be being an awesome beta-tester!

Reply:
Thanks for the explanation - in the mapi-models project I discarded the use of version.sbt (in favour of the user defining this via the github release UI) so it's good to see how the release process can integrate with this file (which is a mechanism that feels neater).

Reply:
Yeah, it's an interesting choice to make, whether to initiate by creating a GitHub release, which is beautiful and elegant, but requires there to _already_ be a tag, with some random human-chosen version number, or to initiate by running the workflow with workflow-dispatch, which is not quite as beautiful, but doesn't require there to be an already-existing tag... I've gone with the second one!